_targets.R

```diff
@@ -0,0 +1,295 @@
+  # we classify w/ logicals based on whether value is greater than or equal to RPs: 2,5,20
+  # also flag
+  tar_target(
+    name = google_historical_class,
+    command = classify_google_historical_data(
+      historical = google_historical,
+      rp_df = gauge_google_wb$return_period
+    )
+  ),
+  # take the classified now class data and just add the basin id's
+  tar_target(
+    name = google_historical_basin,
+    command = gauges_basin_google %>%
+      map(
+        \(basin_level_df){
+          basin_level_df %>%
+            st_drop_geometry() %>%
+            left_join(
+              google_historical_class,
+              by = "gauge_id"
+            )
+        }
+      )
+  ),
+  # might want to turn this into a function so that we can easily change the +/- days parameter
+  # Overview: 
+    # - find all distinct gauge+basin+date+discharge records where threshold (RP2) was exceeded
+    # - find filter to those basin + dates in full historical and scan +/- n (5) days in each basin
+    # - count the number of other gauges in the basin that crossed in that period (n)
+  
+  tar_target(
+    name = ghistorical_rp2_breach_pct_basin_gauges,
+    command = google_historical_basin %>%
+      map(\(dft){
+        # a little wrangling - should do in earlier step an rm from here later
+        dft_c <- dft %>%
+          rename(
+            hybas_id = HYBAS_ID
+          ) %>%
+          mutate(
+            hybas_id = as.character(hybas_id)
+          )
+        # find all distinct flagging events (when RP 2 was exceeded)
+        distinct_flags <- dft_c %>%
+          filter(rp_2_flag) %>%
+          distinct(
+            hybas_id,
+            gauge_id,
+            time
+          )
+
+        distinct_flags %>%
+          pmap_dfr(function(...) {
+            current <- tibble(...)
+
+            # filter ncst to +/-x days
+            ncst_filtered <- dft_c %>%
+              filter(
+                time >= current$time - 5,
+                time <= current$time + 5,
+                hybas_id == current$hybas_id
+              )
+            # get gauge count per basin - could grab this externally as well
+            basin_gauge_count <- dft_c %>%
+              group_by(hybas_id,
+                yr = year(time),
+                time
+              ) %>%
+              summarise(
+                num_gauge = n(), .groups = "drop"
+              ) %>%
+              distinct(hybas_id, num_gauge)
+
+            ncst_filtered %>%
+              mutate(
+                date = current$time
+              ) %>%
+              group_by(
+                hybas_id,
+                date,
+                gauge_id,
+                .drop = F,
+              ) %>%
+              summarise(
+                discharge_crossed = any(gte_rp2), .groups = "drop_last"
+              ) %>%
+              summarise(
+                gauges_crossed = sum(discharge_crossed, na.rm = T),
```

Comment:
Here can just calculate the % just my taking the mean, without referencing the # of gauges. I think also makes sense to do that rather than compare to hard coded # because we should take the % of those available.